src/legacy/ui/public/vis/editors/default/agg_param.js

```diff
@@ -87,25 +93,45 @@ uiModules
               // Whenever the value of the parameter changed (e.g. by a reset or actually by calling)
               // we store the new value in $scope.paramValue, which will be passed as a new value to the react component.
               $scope.paramValue = value;
+
+              $scope.setValidity(true);
+              showValidation();
+            }, true);
+
+            $scope.$watch(() => {
+              // The model can become touched either onBlur event or when the form is submitted.
+              return ngModelCtrl.$touched;
+            }, (value) => {
+              if (value === true) {
```

Comment:
Do we need the `=== true` here? Isn't that anyway just a `undefined/false/true` value?

Reply:
Correct. Updated.